### PR TITLE
Added table of contents

### PR DESCRIPTION
**Pull request by Haud**
@ThibaultKlein @paulmiard 

This is my 2 TODOs: Table of Contents as well as a general overview and guide for contributing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,47 @@
...
+	* [Type Declarations](#type-declarations)
+	* [Nil Checking](#nil-checking)
+	* [Enums](#enums)
+* [Best Practices](BestPractices.md)
 
-This is a work in progress.
-If there is anything you are looking for that is not covered here you should refer to [Github's style guide](https://github.com/github/swift-style-guide).
 
-[Link](BestPractices.md) for Prolific's Swift Best Practices document.
+## Overview ##
```
Comment (ThibaultKlein) [inline README.md:20]: When you look at the README it feels that everything falls under Overview. Is that intended? Shouldn't the rules be under a `## Style Guide Rules##` section?

Comment (Haud) [inline README.md:20]: Yea that makes sense. Updated.

**Comment by ThibaultKlein** [comment]
:+1:

**Comment by paulmiard** [comment]
:+1:

**Pull request merged**